include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,533 @@
+
+    BufferInfo* findInfo( vk::Buffer buffer )
+    {
+        if ( auto entry = m_buffers_info.find( buffer ); entry == m_buffers_info.end() )
+        {
+            throw Error{ "Mman: buffer info not found." };
+        } else
+        {
+            return &entry->second;
+        }
+    } // findInfo
+
+    ImageInfo* findInfo( vk::Image image )
+    {
+        if ( auto entry = m_images_info.find( image ); entry == m_images_info.end() )
+        {
+            throw Error{ "Mman: image info not found." };
+        } else
+        {
+            return &entry->second;
+        }
+    } // findInfo
+
+    VmaAllocation* findAllocation( auto target ) { return &findInfo( target )->allocation; }
+
+    void addInfo( vk::Buffer buffer, const BufferInfo& info ) { m_buffers_info.insert( { buffer, info } ); }
+    void addInfo( vk::Image image, const ImageInfo& info ) { m_images_info.insert( { image, info } ); }
+
+    void clearInfo( vk::Buffer buffer ) { m_buffers_info.erase( buffer ); }
+    void clearInfo( vk::Image image ) { m_images_info.erase( image ); }
+
+    OneTimeCommand createCommand() const { return { m_device, m_cmd_pool, m_queue }; }
+
+    VmaDetailedStatistics getTotalStats() const
+    {
+        VmaTotalStatistics stats{};
+        vmaCalculateStatistics( m_vma, &stats );
+
+        return stats.total;
+    } // getTotalStats
+
+    static constexpr VmaAllocationCreateInfo k_buffer_alloc_create_info{
+        .flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT,
+        .usage = VMA_MEMORY_USAGE_AUTO };
+
+    static constexpr VmaAllocationCreateInfo k_image_alloc_create_info{
+        .usage = VMA_MEMORY_USAGE_AUTO,
+        .requiredFlags = VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT };
+
+    static constexpr VmaVulkanFunctions k_vulkan_functions = {
+        .vkGetInstanceProcAddr = &vkGetInstanceProcAddr,
+        .vkGetDeviceProcAddr = &vkGetDeviceProcAddr };
+
+  public:
+    Mman(
+        vkwrap::VulkanVersion version,
+        vk::Instance instance,
+        vk::PhysicalDevice physical_device,
+        vk::Device logical_device,
+        vk::Queue queue,
+        vk::CommandPool cmd_pool )
+        : m_vma{ VK_NULL_HANDLE },
+          m_device{ logical_device },
+          m_queue{ queue },
+          m_cmd_pool{ cmd_pool }
+    {
+
+        VmaAllocatorCreateInfo create_info{
+            .physicalDevice = physical_device,
+            .device = logical_device,
+            .pVulkanFunctions = &k_vulkan_functions,
+            .instance = instance,
+            .vulkanApiVersion = utils::toUnderlying( version ) };
+
+        VkResult result = vmaCreateAllocator( &create_info, &m_vma );
+        vk::resultCheck( vk::Result{ result }, "Mman: allocator creation error." );
+    } // Mman
+
+    ~Mman() { vmaDestroyAllocator( m_vma ); }
+
+    Mman( const Mman& ) = delete;
+    Mman( Mman&& ) = delete;
+    Mman& operator=( const Mman& ) = delete;
+    Mman& operator=( Mman&& ) = delete;
```

Comment:
Are there any reasons why this class is not movable?

Reply:
This is not entirely obvious to me

Reply:
Because Image and Buffer wrappers must call `create` and `destroy` at the same allocator. To ensure security of such calls I've made allocator not movable.